src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt

```diff
@@ -86,33 +86,32 @@ fun processItemDeclarations(
         }
     }
 
-    if (Namespace.Types in ns) {
-        if (scope is RsFile && scope.isCrateRoot && withPrivateImports) {
-            // Rust injects implicit `extern crate std` in every crate root module unless it is
-            // a `#![no_std]` crate, in which case `extern crate core` is injected. However, if
-            // there is a (unstable?) `#![no_core]` attribute, nothing is injected.
-            //
-            // https://doc.rust-lang.org/book/using-rust-without-the-standard-library.html
-            // The stdlib lib itself is `#![no_std]`, and the core is `#![no_core]`
-            when (scope.attributes) {
-                RsFile.Attributes.NONE ->
-                    if (processor.lazy(STD) { scope.findDependencyCrateRoot(STD) }) return true
-
-                RsFile.Attributes.NO_STD ->
-                    if (processor.lazy(CORE) { scope.findDependencyCrateRoot(CORE) }) return true
-
-                RsFile.Attributes.NO_CORE -> Unit
-            }
-        }
-    }
-
     for (speck in itemImports) {
         check(speck.useGroup == null)
         val path = speck.path ?: continue
         val name = speck.nameInScope ?: continue
         if (processMultiResolveWithNs(name, ns, path.reference, processor)) return true
     }
 
+    if (Namespace.Types in ns && scope is RsMod && withPrivateImports) {
+        // Rust injects implicit `extern crate std` in every module unless it is
```

Comment:
Change "in every crate root module" to "in every module" is wrong
The corresponding documentation:

![image](https://user-images.githubusercontent.com/2539310/54888553-03b5cc00-4eaf-11e9-8796-6cffdbd04e17.png)